src/Orchard.Web/Modules/Orchard.Users/Services/PasswordChangedDateUserDataProvider.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using Orchard.ContentManagement;
+using Orchard.Security;
+using Orchard.Security.Providers;
+using Orchard.Users.Models;
+
+namespace Orchard.Users.Services {
+    public class PasswordChangedDateUserDataProvider : BaseUserDataProvider {
```

Comment:
With this provider the auth cookies is considered invalid on pasword changes. Because of this, when we change password we are automatically logged out of the tenant (from all clients, including the one from which we changed password), and have to sign in anew. I prefer this to having the user remain logged in, but it is different than the current UX.

Reply:
Note: in this PR right now (2018/01/18 15.30) I am not firing the LoggedOut event when this happens.